packages/data-stores/src/newsletter-categories/index.ts

```diff
@@ -0,0 +1,63 @@
+import { useQuery, UseQueryResult } from '@tanstack/react-query';
+import request from 'wpcom-proxy-request';
+
+export type NewsletterCategory = {
+	id: number;
+	name: string;
+	slug: string;
+	description: string;
+	parent: number;
+	subscribed?: boolean;
+};
+
+export type NewsletterCategories = {
+	enabled: boolean;
+	newsletterCategories: NewsletterCategory[];
+	error?: string;
+};
+
+type NewsletterCategoryResponse = {
+	enabled: boolean;
+	newsletter_categories: NewsletterCategory[];
+};
+
+export const getNewsletterCategoriesKey = ( siteId?: string | number ) =>
+	[ 'newsletter-categories', siteId ] as const; // Make tuple readonly for better caching
+
+/**
+ * Converts API response to correct data structure.
+ */
+const convertResponse = ( response: NewsletterCategoryResponse ): NewsletterCategories => ( {
+	enabled: response.enabled,
+	newsletterCategories: response.newsletter_categories,
+} );
+
+/**
+ * React hook to fetch newsletter categories for a site.
+ */
+export const useNewsletterCategories = ( {
```

Comment:
This file is basically replicating https://github.com/Automattic/wp-calypso/blob/add/manually-import-categories/client/data/newsletter-categories/use-newsletter-categories-query.tsx#L0-L1 because I can't access it from the package.

Reply:
Moving this here instead: https://github.com/Automattic/wp-calypso/pull/97561